From 021e309f4f32642b8f53e3c8a162c4af150e269e Mon Sep 17 00:00:00 2001
From: Ales Kozumplik <ales@redhat.com>
Date: Tue, 10 Jul 2012 14:07:02 +0200
Subject: [PATCH] preliminary fix for JOB resons in
 solver_describe_decision().

This makes the function return SOLVER_REASON_RESOLVE_JOB for some unit
rules too. Discussion is ongoing with the libsolv upstream whether this is
a good solution.
---
 src/solver.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/solver.c b/src/solver.c
index 68ff49f..16a0ff8 100644
--- a/src/solver.c
+++ b/src/solver.c
@@ -3437,6 +3437,8 @@ solver_describe_decision(Solver *solv, Id p, Id *infop)
     {
       if (infop)
 	*infop = why;
+      if (why >= solv->jobrules && why < solv->jobrules_end)
+	return SOLVER_REASON_RESOLVE_JOB;
       return SOLVER_REASON_UNIT_RULE;
     }
   why = -why;
-- 
1.7.10.2

